opsdroid/connector/matrix/events.py

```diff
@@ -12,50 +14,76 @@
 ]
 
 
-class MatrixStateEvent(Event):
-    """A Generic matrix state event."""
+class GenericMatrixRoomEvent(Event):
+    """A generic matrix room event.
+
+    Used for things which don't have a specific event type or as a base for
+    matrix specific events.
+    """
 
-    def __init__(self, key, content, *args, state_key=None, **kwargs):  # noqa: D107
+    def __init__(self, event_type, content, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.key = key
         self.content = content
+        self.event_type = event_type
+
+    def __repr__(self):  # noqa: D107
+        return f"<GenericMatrixRoomEvent(room_id={self.target}, event_type={self.event_type}, content={self.content})>"
+
+
+class MatrixStateEvent(GenericMatrixRoomEvent):
+    """A Generic matrix state event."""
+
+    def __init__(self, *args, state_key=None, **kwargs):  # noqa: D107
```

Comment:
This is a small ish breaking change for `MatrixStateEvent` as it no longer accepts "key=" as a kwarg (it's still the same positional arg) it's now a more correct "event_type".

Reply:
it broke the database, but I pushed a fix which works either way already.